src/services/syncFeedPerUser.js

```diff
@@ -3,31 +3,24 @@ const {
   findFollowingUserIds,
   findF2UserIds,
   findFollowingTopicByUser,
-  findUnrelatedUserIds
+  findRelatedUserIds,
+  findUnrelatedUserIds,
+  findUnrelatedAnonUserIds
 } = require('../processes/helper/userIdsToProcess');
+const {followMainFeedBroad, unFollowMainFeedBroad} = require('./followMainFeedBroad');
 const {successResponse, errorResponse} = require('../utils');
 const findAnonymousUserId = require('../databases/functions/users/find-anonymous-user-id');
 
-const syncFeedPerUser = async (req, res) => {
-  try {
-    // sample id = 397e2fee-6af1-4551-9aae-29a7826cd173
-    const {userId} = req.params;
-    await syncFeedPerUserProcess(userId);
-
-    return successResponse(res, 'sync data sucesfully', []);
-  } catch (error) {
-    return errorResponse(res, error.toString(), 500);
-  }
-};
-
 const requestStream = async (feedType, clientFeed, offset, limit) => {
-  if (feedType == 'following') {
-    return await clientFeed.following({offset, limit});
-  }
-  if (feedType == 'follower') {
-    return await clientFeed.followers({offset, limit});
+  let result = {};
+  if (feedType === 'following') {
+    result = await clientFeed.following({offset, limit});
+  } else if (feedType === 'follower') {
+    result = await clientFeed.followers({offset, limit});
+  } else {
+    result = await clientFeed.get({offset, limit});
   }
-  return await clientFeed.get({offset, limit});
+  return result;
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The `requestStream` function has been refactored to use a single `result` variable. This is a good change as it reduces the number of return statements and makes the code cleaner. However, there's no error handling in case any of the async operations fail. Consider adding try-catch blocks around these operations.
```diff
  const requestStream = async (feedType, clientFeed, offset, limit) => {
+   try {
      let result = {};
      if (feedType === 'following') {
        result = await clientFeed.following({offset, limit});
      } else if (feedType === 'follower') {
        result = await clientFeed.followers({offset, limit});
      } else {
        result = await clientFeed.get({offset, limit});
      }
      return result;
+   } catch (error) {
+     console.error(`Error while requesting stream: ${error}`);
+     throw error;
+   }
  };
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->